hyperspy/_signals/dielectric_function.py

```diff
@@ -122,6 +122,7 @@ def get_electron_energy_loss_spectrum(self, zlp, t):
         data = ((-1 / self.data).imag * eels_constant(self, zlp, t).data *
                 self.axes_manager.signal_axes[0].scale)
         s = self._deepcopy_with_new_data(data)
+        s.data = s.data.real
```

Comment:
This is fixing yet another bug, maybe mention it in the PR?

Reply:
Sorry, I forgot to mention this in the PR, although it is mentioned in the commit. I run accross the issue while I was writing the test. The deepcopy creates a new dielectric type signal, but we want an EELS spectrum.